packages/proto-rpc/src/ServerRegistry.ts

```diff
@@ -0,0 +1,154 @@
+    private stopped = false
+
+    public async onRequest(rpcMessage: RpcMessage, callContext?: CallContext): Promise<Uint8Array> {
+        if (this.stopped) {
+            return new Uint8Array()
+        }
+        logger.trace(`Server processing request ${rpcMessage.requestId}`)
+        const methodName = rpcMessage.header.method
+        if (methodName === undefined) {
+            throw new UnknownRpcMethod('Header "method" missing from RPC message')
+        }
+        const fn = this.methods.get(methodName) as RegisteredMethod
+        if (!fn) {
+            throw new UnknownRpcMethod(`RPC Method ${methodName} is not provided`)
+        }
+
+        return await promiseTimeout(1000, fn!(rpcMessage.body, callContext ? callContext : new CallContext()))
+    }
+
+    public async onNotification(rpcMessage: RpcMessage, callContext?: CallContext): Promise<Empty> {
+        if (this.stopped) {
+            return {} as Empty
+        }
+        logger.trace(`Server processing notification ${rpcMessage.requestId}`)
+        const methodName = rpcMessage.header.method
+        if (methodName === undefined) {
+            throw new UnknownRpcMethod('Header "method" missing from RPC message')
+        }
+        const fn = this.methods.get(methodName) as RegisteredNotification
+        if (!fn) {
+            throw new UnknownRpcMethod(`RPC Method ${methodName} is not provided`)
+        }
+        return await promiseTimeout(1000, fn!(rpcMessage.body, callContext ? callContext : new CallContext()))
+    }
+
+    public registerRpcMethod<RequestClass extends Parser<RequestType>, ReturnClass extends Serializer<ReturnType>, RequestType, ReturnType>(
+        requestClass: RequestClass,
+        returnClass: ReturnClass,
+        name: string,
+        fn: (rq: RequestType, _context: CallContext) => Promise<ReturnType>
+    ): void {
+        this.methods.set(name, async (bytes: Uint8Array, callContext: CallContext) => {
+            const request = ConversionWrappers.parseWrapper(() => requestClass.fromBinary(bytes))
+            const response = await fn(request, callContext)
+            return returnClass.toBinary(response)
+        })
+    }
+
+    public registerRpcNotification<RequestClass extends Parser<RequestType>, RequestType>(
+        requestClass: RequestClass,
+        name: string,
+        fn: (rq: RequestType, _context: CallContext) => Promise<Empty>
+    ): void {
+        this.methods.set(name, async (bytes: Uint8Array, callContext: CallContext): Promise<Empty> => {
+            const request = ConversionWrappers.parseWrapper(() => requestClass.fromBinary(bytes))
+            const response = await fn(request, callContext)
+            return Empty.toBinary(response)
+        })
+    }
+
+    public stop(): void {
+        this.stopped = true
+        this.methods.clear()
+    }
+}
+
+export class CallContext implements ServerCallContext, ProtoRpcOptions {
```

Comment:
We could move this class to a separate file (if we need the class)

Reply:
Moved to ProtoCallContext